gh-actions/common/run-autopkgtest/action.yml

```diff
@@ -0,0 +1,172 @@
+name: Run autopkgtest for debian packages
+description: |
+  Runs autopkgtest for a built source package, optionally using
+  already built deb files.
+  The build happens in an LXD container so that autopkgtest can setup all the
+  required parameters to run the test with all the expected constraints.
+
+inputs:
+  source-changes:
+    description: |
+      The path to the source .changes file.
+      The the rest of the source files are expected to be in the same directory
+    required: true
+
+  debs-dir:
+    description: |
+      Directory where the deb files are located.
+      It will be the same of the source-dir if not provided.
+    required: false
+
+  lxd-image:
+    description: |
+      The LXD image used to build the package.
+      A docker-style image aliases can also be used.
+    required: false
+    default: ubuntu:rolling
+
+  autopkgtest-args:
+    description: Additional arguments to pass to autopkgtest
+    required: false
+    default: ''
+
+runs:
+  using: "composite"
+  steps:
+    - name: Check prerequisites
+      shell: bash
+      run: |
+        echo "::group::Check prerequisites"
+        set -eux
+
+        sudo apt install dctrl-tools
+        pkg_version=$(grep-dctrl -s Version -n - "${{ inputs.source-changes }}")
+
+        if [ -z "${pkg_version}" ]; then
+          echo "${{ inputs.source-changes }} is not a valid source .changes file" | \
+            tee "${GITHUB_STEP_SUMMARY}"
+          exit 1
+        fi
+
+        echo "PKG_VERSION=${pkg_version}" >> "${GITHUB_ENV}"
+
+        debs_dir=${{ inputs.debs-dir }}
+        if [ -z "$debs_dir" ]; then
+          debs_dir=$(dirname "${{ inputs.source-changes }}")
+        fi
+
+        if ls "${debs_dir}"/*_"${pkg_version}"_*.deb &>/dev/null; then
+          echo "DEBS_DIR=$debs_dir" >> "${GITHUB_ENV}"
+          echo "Testing debian packages"
+        else
+          msg="${debs_dir} is not containing any debian package"
+          echo "${msg}"
+
+          if [[ "$debs_dir" == "${{ inputs.debs-dir }}" ]]; then
+            echo "${msg}" >> "${GITHUB_STEP_SUMMARY}"
+            exit 1
+          fi
+
+          echo "The package source ${{ inputs.source-changes }} will be compiled before being tested."
+        fi
+
+        echo "::endgroup::"
+
+    - name: Setup LXD
+      uses: canonical/setup-lxd@v0.1.2
+      with:
+        channel: latest/stable
```

Comment:
any reasons we pick the channel ourself and not keep the default?

Reply:
Oh... It defaults to candidate, so I thought it was better to be a bit more conservative there

https://github.com/canonical/setup-lxd/blob/54a5806e490d92e207b57183cf111ed54bbdeff4/action.yml#L5

Reply:
making sense